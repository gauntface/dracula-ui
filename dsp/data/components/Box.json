{
  "class": "component",
  "type": "page",
  "id": "Box",
  "name": "Box",
  "last_updated": "2021-03-15T05:45:09.966Z",
  "last_updated_by": "System",
  "description": "Box is the most primitive component of Dracula UI.\nUsing Box allows for consumers of the components library to compose\nmore complex patterns, components, and UIs.\n\nBox includes built-in Color and Spacing properties that make building\ncomplex components convenient and consistent.",
  "related_entity_ids": [],
  "tags": ["component"],
  "snippets": {
    "trigger": "box",
    "languages": {
      "javascriptreact": "<Box color=\"purple\">Dracula</Box>;\n",
      "html": "<div color=\"purple\" class=\"drac-box drac-bg-purple\">Dracula</div>\n"
    }
  },
  "ext_com_draculaui_variations": [
    {
      "title": "Usage",
      "html": "<div color=\"purple\" class=\"drac-box drac-bg-purple\">Dracula</div>\n",
      "react": "<Box color=\"purple\">Dracula</Box>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Box.png",
      "svgData": "./dsp/assets/svgs/Box.svg"
    },
    {
      "title": "Colors",
      "html": "<div color=\"pinkPurple\" class=\"drac-box drac-bg-pink-purple\">Pink Purple</div>\n",
      "react": "<Box color=\"pinkPurple\">Pink Purple</Box>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxColors.png",
      "svgData": "./dsp/assets/svgs/BoxColors.svg"
    },
    {
      "title": "Rounded",
      "html": "<div style=\"display: flex; flex-direction: row\">\n  <div\n    color=\"purple\"\n    rounded=\"none\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-none\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"sm\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-sm\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"base\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"lg\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-lg\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"2xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-2xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"3xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-3xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"full\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-full\"\n  ></div>\n</div>\n",
      "react": "<div style={{ display: \"flex\", flexDirection: \"row\" }}>\n  <Box\n    color=\"purple\"\n    rounded=\"none\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"sm\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"base\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"lg\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"2xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"3xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"full\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxRounded.png",
      "svgData": "./dsp/assets/svgs/BoxRounded.svg"
    },
    {
      "title": "Generic Elements",
      "html": "<div>\n  <button as=\"button\" color=\"pinkPurple\" class=\"drac-box drac-bg-pink-purple\">\n    Button</button\n  ><a as=\"a\" color=\"green\" href=\"#\" class=\"drac-box drac-bg-green\">Link</a>\n</div>\n",
      "react": "<div>\n  <Box as=\"button\" color=\"pinkPurple\">\n    Button\n  </Box>\n  <Box as=\"a\" color=\"green\" href=\"#\">\n    Link\n  </Box>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxGeneric Elements.png",
      "svgData": "./dsp/assets/svgs/BoxGeneric Elements.svg"
    }
  ],
  "ext_com_draculaui_props": {
    "color": {
      "defaultValue": null,
      "description": "The background color.",
      "name": "color",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": {
        "name": "\"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"white\" | \"black\" | \"blackSecondary\" | \"blackLight\" | undefined"
      }
    },
    "glowColor": {
      "defaultValue": null,
      "description": "The glow color.",
      "name": "glowColor",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": {
        "name": "\"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | undefined"
      }
    },
    "rounded": {
      "defaultValue": null,
      "description": "The border radius.",
      "name": "rounded",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"base\" | \"lg\" | \"xl\" | \"2xl\" | \"3xl\" | \"full\" | undefined"
      }
    },
    "as": {
      "defaultValue": null,
      "description": "The HTML element to be used",
      "name": "as",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
    },
    "p": {
      "defaultValue": null,
      "description": "",
      "name": "p",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
      }
    },
    "py": {
      "defaultValue": null,
      "description": "",
      "name": "py",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
      }
    },
    "px": {
      "defaultValue": null,
      "description": "",
      "name": "px",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
      }
    },
    "pt": {
      "defaultValue": null,
      "description": "",
      "name": "pt",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
      }
    },
    "pb": {
      "defaultValue": null,
      "description": "",
      "name": "pb",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
      }
    },
    "pl": {
      "defaultValue": null,
      "description": "",
      "name": "pl",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
      }
    },
    "pr": {
      "defaultValue": null,
      "description": "",
      "name": "pr",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
      }
    },
    "m": {
      "defaultValue": null,
      "description": "",
      "name": "m",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
      }
    },
    "my": {
      "defaultValue": null,
      "description": "",
      "name": "my",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
      }
    },
    "mx": {
      "defaultValue": null,
      "description": "",
      "name": "mx",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
      }
    },
    "mt": {
      "defaultValue": null,
      "description": "",
      "name": "mt",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
      }
    },
    "mb": {
      "defaultValue": null,
      "description": "",
      "name": "mb",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
      }
    },
    "ml": {
      "defaultValue": null,
      "description": "",
      "name": "ml",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
      }
    },
    "mr": {
      "defaultValue": null,
      "description": "",
      "name": "mr",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
      }
    }
  },
  "ext_com_draculaui_docgen": {
    "tags": {},
    "description": "Box is the most primitive component of Dracula UI.\nUsing Box allows for consumers of the components library to compose\nmore complex patterns, components, and UIs.\n\nBox includes built-in Color and Spacing properties that make building\ncomplex components convenient and consistent.",
    "displayName": "Box",
    "methods": [],
    "props": {
      "color": {
        "defaultValue": null,
        "description": "The background color.",
        "name": "color",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": {
          "name": "\"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"white\" | \"black\" | \"blackSecondary\" | \"blackLight\" | undefined"
        }
      },
      "glowColor": {
        "defaultValue": null,
        "description": "The glow color.",
        "name": "glowColor",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": {
          "name": "\"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | undefined"
        }
      },
      "rounded": {
        "defaultValue": null,
        "description": "The border radius.",
        "name": "rounded",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"base\" | \"lg\" | \"xl\" | \"2xl\" | \"3xl\" | \"full\" | undefined"
        }
      },
      "as": {
        "defaultValue": null,
        "description": "The HTML element to be used",
        "name": "as",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
      },
      "p": {
        "defaultValue": null,
        "description": "",
        "name": "p",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
        }
      },
      "py": {
        "defaultValue": null,
        "description": "",
        "name": "py",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
        }
      },
      "px": {
        "defaultValue": null,
        "description": "",
        "name": "px",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
        }
      },
      "pt": {
        "defaultValue": null,
        "description": "",
        "name": "pt",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
        }
      },
      "pb": {
        "defaultValue": null,
        "description": "",
        "name": "pb",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
        }
      },
      "pl": {
        "defaultValue": null,
        "description": "",
        "name": "pl",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
        }
      },
      "pr": {
        "defaultValue": null,
        "description": "",
        "name": "pr",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
        }
      },
      "m": {
        "defaultValue": null,
        "description": "",
        "name": "m",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
        }
      },
      "my": {
        "defaultValue": null,
        "description": "",
        "name": "my",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
        }
      },
      "mx": {
        "defaultValue": null,
        "description": "",
        "name": "mx",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
        }
      },
      "mt": {
        "defaultValue": null,
        "description": "",
        "name": "mt",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
        }
      },
      "mb": {
        "defaultValue": null,
        "description": "",
        "name": "mb",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
        }
      },
      "ml": {
        "defaultValue": null,
        "description": "",
        "name": "ml",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
        }
      },
      "mr": {
        "defaultValue": null,
        "description": "",
        "name": "mr",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"lg\" | \"xxs\" | \"xs\" | \"md\" | undefined"
        }
      }
    }
  }
}
