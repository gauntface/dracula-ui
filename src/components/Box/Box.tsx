import cx from 'classnames/dedupe'
import React, { AllHTMLAttributes } from 'react'
import { colors } from '../../base/colors'
import { spacingClasses, SpacingPropType } from '../../base/spacing'

type Element = HTMLElementTagNameMap

export const roundedBorders = {
  none: 'drac-rounded-none',
  sm: 'drac-rounded-sm',
  base: 'drac-rounded',
  lg: 'drac-rounded-lg',
  xl: 'drac-rounded-xl',
  '2xl': 'drac-rounded-2xl',
  '3xl': 'drac-rounded-3xl',
  full: 'drac-rounded-full',
}

/**
 * Box Props
 */
export interface BoxProps<K extends keyof Element = 'div'>
  extends AllHTMLAttributes<K> {
  /** The background color. */
  color?: keyof typeof colors

  /** The border radius. */
  rounded?: keyof typeof roundedBorders

  /** Dracula UI standard spacing properties. */
  spacing?: SpacingPropType

  /** The HTML element to be used */
  as?: K
}

/**
 * Box is the most primitive component of Dracula UI.
 * Using Box allows for consumers of the components library to compose
 * more complex patterns, components, and UIs.
 *
 * Box includes built-in Color and Spacing properties that make building
 * complex components convenient and consistent.
 */
export function Box<T extends keyof Element>(props: BoxProps<T>) {
  const finalProps = {
    ...props,
    className: cx(
      `drac-box`,
      props.color && colors[props.color],
      props.rounded && roundedBorders[props.rounded],
      ...spacingClasses(props.spacing),
      props.className
    )
  }

  const as = finalProps.as ?? 'div'
  return React.createElement(as, finalProps, props.children)
}

Box.displayName = 'Box'
