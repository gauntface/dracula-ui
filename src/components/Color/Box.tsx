import React, { HTMLAttributes } from 'react'

export const colors = {
  black: 'drac-bg-black',
  white: 'drac-bg-white',
  cyan: 'drac-bg-cyan',
  green: 'drac-bg-green',
  orange: 'drac-bg-orange',
  pink: 'drac-bg-pink',
  purple: 'drac-bg-purple',
  red: 'drac-bg-red',
  yellow: 'drac-bg-yellow',
  purpleCyan: 'drac-bg-purple-cyan',
  yellowPink: 'drac-bg-yellow-pink',
  cyanGreen: 'drac-bg-cyan-green',
  pinkPurple: 'drac-bg-pink-purple',
  animated: 'drac-bg-animated'
}

export interface BoxProps extends HTMLAttributes<HTMLDivElement> {
  color?: keyof typeof colors
}

export const Box: React.FC<BoxProps> = props => {
  const finalProps = {
    ...props,
    className: `drac-box ${colors[props.color ?? 'black']}`
  }

  return <div {...finalProps} />
}
