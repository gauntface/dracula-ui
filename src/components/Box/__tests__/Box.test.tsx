import React from 'react'
import * as ReactDOM from 'react-dom'
import { Box } from '@/components/Color/Box'

describe('Box', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Box color="white" />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})
