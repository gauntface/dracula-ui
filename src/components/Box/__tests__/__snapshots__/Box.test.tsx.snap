// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Box Basic Usage 1`] = `
Object {
  "docs": "",
  "html": "<div color=\\"purple\\" class=\\"drac-box drac-bg-purple\\">Dracula</div>
",
  "react": "<Box color=\\"purple\\">Dracula</Box>;
",
  "title": "Usage",
}
`;

exports[`Site: Box Colors 1`] = `
Object {
  "docs": "",
  "html": "<div color=\\"pinkPurple\\" class=\\"drac-box drac-bg-pink-purple\\">Pink Purple</div>
",
  "react": "<Box color=\\"pinkPurple\\">Pink Purple</Box>;
",
  "title": "Colors",
}
`;

exports[`Site: Box Generic Elements 1`] = `
Object {
  "docs": "",
  "html": "<div>
  <button
    is=\\"button\\"
    color=\\"pinkPurple\\"
    classname=\\"drac-box drac-bg-pink-purple\\"
  >
    Button</button
  ><a is=\\"a\\" color=\\"green\\" href=\\"#\\" classname=\\"drac-box drac-bg-green\\">Link</a>
</div>
",
  "react": "<div>
  <Box is=\\"button\\" color=\\"pinkPurple\\">
    Button
  </Box>
  <Box is=\\"a\\" color=\\"green\\" href=\\"#\\">
    Link
  </Box>
</div>;
",
  "title": "Generic Elements",
}
`;
